core/storage/question/gae_models.py

```diff
@@ -365,51 +362,67 @@ def get_question_skill_links_based_on_difficulty_equidistributed_by_skill(
             # We fetch more questions here in order to try and ensure that the
             # eventual number of returned questions is sufficient to meet the
             # number requested, even after deduplication.
-            new_question_skill_link_models = (
-                equal_questions_query.fetch(question_count_per_skill * 2))
+            new_question_skill_link_models = equal_questions_query.fetch()
             for model in new_question_skill_link_models:
                 if model.question_id in question_skill_link_mapping:
                     new_question_skill_link_models.remove(model)
 
-            if len(new_question_skill_link_models) < question_count_per_skill:
+            if len(new_question_skill_link_models) >= question_count_per_skill:
+                new_question_skill_link_models = utils.get_random_sample(
+                    new_question_skill_link_models, question_count_per_skill)
+            else:
                 # Fetch QuestionSkillLinkModels with difficulty smaller than
                 # requested difficulty and sort them by decreasing difficulty.
                 easier_questions_query = query.filter(
                     cls.skill_difficulty < difficulty_requested)
                 easier_questions_query = easier_questions_query.order(
```

Comment:
I guess we dont need this ordering anymore right

Reply:
Done.